src/NServiceBus.Core.Tests/Recoverability/When_message_is_deferred_by_slr.cs

```diff
@@ -0,0 +1,145 @@
+﻿namespace NServiceBus.Core.Tests.Recoverability
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Threading.Tasks;
+    using NServiceBus.Pipeline;
+    using NServiceBus.Transports;
+    using NUnit.Framework;
+    using Testing;
+
+    [TestFixture]
+    public class When_message_is_deferred_by_slr
+    {
+        FakeDispatchPipeline dispatchPipeline;
+
+        [SetUp]
+        public void SetUp()
+        {
+            dispatchPipeline = new FakeDispatchPipeline();
+        }
+
+        [Test]
+        public async Task Should_abort_message_receive_when_marking_for_deferral()
+        {
+            var chain = CreateBehaviorChain(new FailingBehavior());
+
+            var context = CreateContext();
+
+            await chain.Invoke(context);
+
+            Assert.IsTrue(context.ReceiveOperationAborted, "Message receive should be aborted by SecondLevelRetries");
+        }
+
+        [Test]
+        public async Task Should_schedule_delayed_delivery_for_failed_message()
+        {
+            var failingBehavior = new FailingBehavior();
+
+            var chain = CreateBehaviorChain(failingBehavior);
+
+            var message = CreateMessage();
+
+            await chain.Invoke(CreateContext(message));
+            await chain.Invoke(CreateContext(message));
```

Comment:
Shouldn't we assert that the receive operation was aborted before the second call to `Invoke` or are we relying on the previous test to assert it?